File: vol1/ch5/LeetCode219.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums,int k) {
        unordered_map<int,int> hmap;
        int n=nums.size();
        for (int i=0;i<n;i++) {
            int d=nums[i];
            if (hmap.count(d)==1 && i-hmap[d]<=k)
                return true;
            else
                hmap[d]=i;
        }
        return false;
    }
};

File: vol1/ch5/LeetCode219.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hmap={}                                 #定义一个字典
        for i,d in enumerate(nums):
            if d in hmap and i-hmap[d]<=k:      #找到后返回结果
                return True
            else:
                hmap[d]=i                       #添加到hmap
        return False
